lib/hmmlearn/base.py

```diff
@@ -549,30 +558,45 @@ def _do_viterbi_pass(self, log_frameprob):
             log_frameprob)
         return log_prob, state_sequence
 
-    def _do_forward_scaling_pass(self, frameprob):
+    def _do_forward_scaling_pass(self, frameprob, startprob=None,
```

Comment:
Should these simply be transformed into private free functions to which you *always* have to pass startprob and transmat?  Ditto below.

Reply:
Let me try it and we can see what it looks like.

Reply:
Should be cleaned up now - we've since simplified this